src/components/Courses.js

```diff
@@ -41,88 +52,73 @@ const useStyles = makeStyles((theme) => ({
     justifyContent: 'center',
   },
   container: {
+    marginTop: '2em',
+  },
+  waitlistContainer: {
     marginTop: '1em',
   },
   actions: {
     flexWrap: 'wrap',
     justifyContent: 'center',
+    height: 45,
   },
 }))
 
-export default function Courses(props) {
+const Course = ({ classes, courses }) => {
+  return courses.map((course) => {
+    return (
+      <div className={classes.courseContainer} key={course.crn}>
+        <div className={classes.container}>
+          <Card className={classes.card}>
+            <CourseHeader course={course} />
+            <CardContent className={classes.content}>
+              <div
+                className={course.waitlist === '0' ? classes.container : classes.waitlistContainer}
+              >
+                <Meetings meetings={course.meetings} />
+              </div>
+            </CardContent>
+            <CardActions className={classes.actions}>
+              <CourseDetails course={course} />
+              <Instructors instructors={course.instructors} />
+            </CardActions>
+          </Card>
+        </div>
+      </div>
+    )
+  })
+}
+
+export default function Courses() {
   const classes = useStyles()
   const courses = useSelector((state) => state.courses)
-  const courses_fetched = useSelector((state) => state.fetched)
   const courses_fetching = useSelector((state) => state.fetching)
   const courses_error = useSelector((state) => state.error)
+  const courses_fetched = useSelector((state) => state.fetched)
 
-  if (courses_fetching === true) {
+  if (courses_fetched && !courses_error) {
     return (
-      <div className={classes.loading}>
-        <CircularProgress color='secondary' size={50} />
+      <div className={classes.root}>
+        <Course courses={courses} classes={classes} />
       </div>
     )
-  } else if (courses_fetched === true && courses_error === true) {
+  } else if (courses_fetched && courses_error) {
     return (
       <div className={classes.error}>
         <ErrorMessages />
       </div>
     )
-  } else if (courses === null || courses === []) {
+  } else if (courses_fetching) {
+    return (
+      <div className={classes.loading}>
+        <CircularProgress color='secondary' size={50} />
+      </div>
+    )
+  } else {
     return (
       <Typography variant='h3' className={classes.empty} tabIndex='0'>
         You currently have no courses for this semester.
```

Comment:
Currently there is no scenario where this element would be shown. To get to this point in the conditional `courses_fetched` and `courses_fetching` must both be false, which is an impossible scenario. This should probably check if courses is an empty array.

Reply:
Good idea, i'll fix this